ReportingServicesTools/Functions/CatalogItems/Rest/Start-RsRestCacheRefreshPlan.ps1

```diff
@@ -0,0 +1,130 @@
+            and refreshes them.
+
+        .EXAMPLE
+            Start-RsReportRefresh -RsReport "/MyReport" -WebSession $session
+
+            Description
+            -----------
+            Fetches the CacheRefreshPlans of a report named "MyReport" catalog item found in "/" folder from the Report Server located at specificed WebSession object, 
+            and refreshes them.
+
+        .EXAMPLE
+            Start-RsReportRefresh -RsReport "/MyReport" -ReportPortalUri http://myserver/reports
+
+            Description
+            -----------
+            Fetches the CacheRefreshPlans of a report named "MyReport" found in "/" folder from the Report Server located at http://myserver/reports, 
+            and refreshes them.
+    #>
+
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+    param(
+        [Parameter(Mandatory = $false, ValueFromPipelineByPropertyName = $true)]
+        [Alias('ItemPath','Path', 'RsItem')]
+        [string]
+        $RsReport,
+
+        [Parameter(Mandatory = $false, ValueFromPipelineByPropertyName = $true)]
+        [Alias('CacheRefreshPlan')]
+        [string]
+        $Id = $null,
+
+        [Parameter(Mandatory = $false, ValueFromPipelineByPropertyName = $true)]
+        [string]
+        $ReportPortalUri,
+
+        [Alias('ApiVersion')]
+        [ValidateSet("v2.0")]
+        [string]
+        $RestApiVersion = "v2.0",
+
+        [Alias('ReportServerCredentials')]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Microsoft.PowerShell.Commands.WebRequestSession]
+        $WebSession
+    )
+    Begin
+    {
+        $WebSession = New-RsRestSessionHelper -BoundParameters $PSBoundParameters
+        $ReportPortalUri = Get-RsPortalUriHelper -WebSession $WebSession
+        $CacheRefreshPlansUri = $ReportPortalUri + "api/$RestApiVersion/CacheRefreshPlans({0})/Model.Execute"
+    }
+    Process
+    {
+        try
+        {
+            if (-not $RsReport)
+            {
+                Write-Verbose "Fetching CacheRefreshPlans for Id $Id..."
+                $CacheRefreshPlansUri = [String]::Format($CacheRefreshPlansUri, $Id)
+            }
+            else
+            {
+                Write-Verbose "Fetching CacheRefreshPlans for $RsReport..."
+                if ($Credential -ne $null)
+                {
+                    $RefreshPlan = Get-RsCacheRefreshPlan -ReportPortalUri $ReportPortalUri -RsReport $RsReport -WebSession $WebSession -Credential $Credential -Verbose:$false
+                }
+                else
+                {
+                    $RefreshPlan = Get-RsCacheRefreshPlan -ReportPortalUri $ReportPortalUri -RsReport $RsReport -WebSession $WebSession -Verbose:$false
+                }
+                $CacheRefreshPlansUri = [String]::Format($CacheRefreshPlansUri, $RefreshPlan.Id)
```

Comment:
does it work with more than one cache refresh plan (if the report has more than one?

Reply:
It did, but it formats the results better now.